### PR TITLE
feat(share-video) Add capability for sharing any direct link video

### PR DESCRIPTION
**Pull request by quitrk**

### CODE REVIEW

**Comment by quitrk** [comment]
Updated tests here https://github.com/jitsi/jitsi-meet-torture/pull/381/files

**Comment by muscat1** [comment]
Took this for a test run and it runs pretty well, I'll start taking a deeper look into the code, but before that, a couple of notes:

1. To me, the wording "direct video link" seemed a bit obscure, not sure if a user will be able to tell that they need to link an mp4 hosted somewhere, rather than, for instance, a vimeo link (that won't work)

2. The avatar for such a video looks strange (not sure if the "Video" should overlap the initial)

Screenshot 2021-05-12 at 12 13 22

3. The "un-share" label should probably reflect the fact that you're stopping a generic share, rather than a YouTube share. (Maybe something like "Stop video"?)

Screenshot 2021-05-12 at 12 13 02

File: modules/API/API.js [hunk 1]
```diff
@@ -263,6 +264,18 @@ function initCommands() {
...
 
+        'start-share-video': url => {
+            logger.debug('Share video command received');
+            sendAnalytics(createApiEvent('share.video.start'));
+            APP.store.dispatch(playSharedVideo(url));
+        },
+
+        'stop-share-video': () => {
+            logger.debug('Share video command received');
+            sendAnalytics(createApiEvent('share.video.start'));
```
Comment (muscat1) [inline modules/API/API.js:275]: `shared.video.stop` I'm guessing :D

File: react/features/shared-video/components/web/SharedVideo.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+class SharedVideo extends Component<Props> {
+    /**
+     * Computes the width and the height of the component.
+     *
+     * @returns {{
+     *  height: number,
+     *  width: number
+     * }}
+     */
+    getDimmensions() {
```
Comment (muscat1) [inline react/features/shared-video/components/web/SharedVideo.js:61]: Nit: typo

**Comment by muscat1** [comment]
Took a brief look over the entire thing and it looks good to me, just a couple of minor nits. Kudos for moving the whole shared video into react!
I'm curious if this plays well with the native app, maybe @saghul can also cast a glance.

**Comment by Calinteodor** [comment]
I managed to test this using xcode simulator. 

1. When sharing a youtube video, if you decide to stop it, the video stops but the iframe and video participant avatar thumbnail don't disappear from the ongoing conference.

2. I don't know if sharing a direct video link will be a feature available for native as well, but the placeholder from the SharedVideoDialog suggests that. 

3. I also tried a direct video link on the simulator, but it doesn't work.
 
https://user-images.githubusercontent.com/26413496/117961810-350be300-b327-11eb-9c0b-c28d434f9226.mov



https://user-images.githubusercontent.com/26413496/117961796-2fae9880-b327-11eb-8867-26c6b96a1296.mov

**Comment by quitrk** [comment]
Addressed the issues on native app. For the scope of this PR, I'm adding just the web functionality, I'll work on mobile in a subsequent PR

**Review by muscat1 — APPROVED** [review]

**Pull request merged**